readthedocs/projects/tasks/search.py

```diff
@@ -14,6 +14,193 @@
 log = structlog.get_logger(__name__)
 
 
+class Indexer:
+
+    """
+    Base class for doing operations over each file from a build.
+
+    The process method should be implemented to apply the operation
+    over each file, and the collect method should be implemented
+    to collect the results of the operation after processing all files.
+
+    `sync_id` is used to differentiate the files from the current sync from the previous one.
+    """
+
+    def process(self, html_file: HTMLFile, sync_id: int):
+        raise NotImplementedError
+
+    def collect(self, sync_id: int):
+        raise NotImplementedError
+
```

Comment:
I wonder if this should be called `save` or `store`? `collect` isn't a super common usage here.

Reply:
I chose collect as this "collects the results", that meaning anything. Can also be `post_process`, but I'm fine with anything.